ignite/_utils.py

```diff
@@ -11,41 +11,14 @@ def _to_hours_mins_secs(time_taken):
     return hours, mins, secs
 
 
-def to_variable(batch, cuda=False, volatile=False):
-    if torch.is_tensor(batch):
-        if cuda:
-            batch = batch.cuda()
-        return Variable(batch, volatile=volatile)
-    elif isinstance(batch, string_classes):
-        return batch
-    elif isinstance(batch, collections.Mapping):
-        return {k: to_variable(sample, cuda=cuda, volatile=volatile) for k, sample in batch.items()}
-    elif isinstance(batch, collections.Sequence):
-        return [to_variable(sample, cuda=cuda, volatile=volatile) for sample in batch]
-    else:
-        raise TypeError(("batch must contain tensors, numbers, dicts or lists; found {}"
-                         .format(type(batch[0]))))
-
-
-def to_tensor(variable, cpu=False):
-    if isinstance(variable, Variable):
-        t = variable.data
-        if cpu:
-            t = t.cpu()
-        return t
-    elif isinstance(variable, string_classes):
-        return variable
-    elif isinstance(variable, collections.Mapping):
-        return {k: to_tensor(sample, cpu=cpu) for k, sample in variable.items()}
-    elif isinstance(variable, collections.Sequence):
-        return [to_tensor(sample, cpu=cpu) for sample in variable]
-    else:
-        raise TypeError(("Argument variable must contain torch.autograd.Variable, numbers, dicts or lists; found {}"
-                         .format(type(variable))))
+def convert_tensor(tensor, requires_grad=False, device=None):
+    if device:
```

Comment:
@sean-adler just remarked that you removed the multiple `if-elif-else` tree as it was in `to_variable`. I think we should keep it. Otherwise you implicitly imposes that batch is composed of only tensors and not list of tensors or a dict of tensors etc.